### ISSUE TITLE
Questions: one about urls, one about models

### ISSUE THREAD

**Issue by SalahAdDin** [torchbox/wagtaildemo#47]
Hi guys, 

I have finally completed my project with django wagtail is a great tool, I love it, but I have two questions to finish the job properly.

First, I need to get the url of two pages, or they may be, but given finds its slug or title, nothing else, this is so that the url is global, the base.html I want to put in, and follow the same in all pages, but not how to do this.

I have a problem when calling the feed_image of a standard page from a carousel. In my template I call as follows: ```carousel_item.link_page.feed_image ```, as you can see, the artifact that relates to my page, which is in fact the home page is a standard page, which in turn has a feed_image, are several related to the home page, that is, several devices in the carousel, and when put in the template, I can access all areas of the page, with excepsión of feed_image, I have not checked whether the field search_index have that problem, but with this field have feed_image problem and do not know how to fix it, I need to bring that image to bring it into the template, to make a menu with pictures, but so far I could.

It is very urgent to solve these problems.

Thank you very much for everything friend. Thanks for the show.

**Comment by zerolab** [comment]
Hi @SalahAdDin, 

I believe you are missing `specific()` (see [docs](http://docs.wagtail.io/en/latest/core_components/pages/creating_pages.html?highlight=specific#wagtail.wagtailcore.models.Page.specific)). That is call `carousel_item.link_page.specific.feed_image`. This will give you access to all fields in link_page.

Hope this helps.

**Comment by gasman** [comment]
@SalahAdDin Sorry, I don't understand your first question - please can you rephrase this? (Or perhaps this is answered by my reply here https://groups.google.com/d/msg/wagtail/zzCqYPOKfPU/8PbXGpj4GhIJ ?)

**Comment by SalahAdDin** [comment]
@gasman , don't worry man, @zerolab help me with the first question. @zerolab Thank you very much!

**Comment by gasman** [comment]
OK. The first question to ask is: are you sure you need a variable for this? If your 'about us' page has the slug 'about-us' and you know that this will never change, then there's nothing wrong with simply hard-coding the link in the template: `<a href="/about-us/">About us</a>`. (If you think that the page might change its name in future, then you'll need to decide how your code will find the correct page... in the example below, I'm assuming that you've given it a unique page type, AboutUsPage.)

The easiest way to define a variable to be available on all templates (including base.html) is to create a [context processor](https://docs.djangoproject.com/en/1.7/ref/templates/api/#subclassing-context-requestcontext). Here's an example:

In a new file `core/context_processors.py`:

    from core.models import AboutUsPage, ContactsPage

    def global_vars(request):
        return {
            'about_us_page': AboutUsPage.objects.get(),
            'contacts_page': ContactsPage.objects.get(),
        }

In `settings/base.py`, add `'core.context_processors.global_vars'` to `TEMPLATE_CONTEXT_PROCESSORS`:

    from django.conf import global_settings
    TEMPLATE_CONTEXT_PROCESSORS = global_settings.TEMPLATE_CONTEXT_PROCESSORS + (
        'django.core.context_processors.request',
        'core.context_processors.global_vars',
    )

The variables `about_us_page` and `contacts_page` are now available on all of your templates, and you can write something like:

    {% load wagtailcore_tags %}
    <ul>
        <li><a href="{% pageurl about_us_page %}">About us</a></li>
        <li><a href="{% pageurl contacts_page %}">Contacts</a></li>
    </ul>

**Comment by SalahAdDin** [comment]
Oh, thank you.
I have a question, both page are standard page, for this reason, i think that is needed filter by slug, some like ```'slug'=='AboutUs'``` or ```'slug'=='ContactUs'```, right?

**Comment by gasman** [comment]
Actually, I forgot about the `slugurl` custom tag http://docs.wagtail.io/en/latest/core_components/pages/writing_templates.html#slugurl - this provides a much easier solution.

    {% load wagtailcore_tags %}
    <a href="{% slugurl 'AboutUs' %}">About us</a>

**Comment by SalahAdDin** [comment]
ok XD
Yes, slugurl helpme, thank you man, thank you very much.

I love this app, wagtail is the better, and his community too.

Thank you.
